meteor/client/ui/SegmentTimeline/Parts/FlattenedSourceLayers.tsx

```diff
@@ -58,6 +58,7 @@ export function FlattenedSourceLayers(props: IFlattenedSourceLayerProps): JSX.El
 									onDoubleClick={props.onPieceDoubleClick}
 									mediaPreviewUrl={props.mediaPreviewUrl}
 									piece={piece}
+									pieces={layer.pieces.map((p) => p.instance.piece)}
```

Comment:
Using `map` here will cause all the `SourceLayerItemContainer` child components to be rerendered whenever this `FlattenedSourceLayers` component does. It would be good to use some `useMemo` to avoid this

Reply:
Ok, I've moved that to the top level to construct a map of Pieces https://github.com/nrkno/sofie-core/pull/890/commits/2e3d8ea1562b7418f76dbaf47c95cf281f14c1d4
The `?? []` feels unnecessary but I don't like the idea of `piecesForLayers.get(...)!` much better